src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/schema/index.ts

```diff
@@ -48,20 +51,28 @@ export const _lensApiStateSchema: any = schema.oneOf([
   legacyMetricStateSchema,
   xyStateSchema,
   gaugeStateSchema,
+  heatmapStateSchema,
   tagcloudStateSchema,
   // disable for now to avoid type issues at the config builder root level
   // TODO: enabled once transformations are available
   // partitionStateSchema,
 ]);
 
-export type LensApiState = MetricState | LegacyMetricState | GaugeState | TagcloudState;
+export type LensApiState =
+  | MetricState
+  | LegacyMetricState
+  | GaugeState
+  | XYState
```

Comment:
The missing `XYState` was causing many of the `@ts-expect-error` issues removed in this pr.